app/services/katello/pulp3/migration_plan.rb

```diff
@@ -105,7 +105,15 @@ def target_link_map
       end
 
       def name_for_content_view(content_view, root_repo)
-        "#{content_view.label}-#{root_repo.label}"
+        name = "#{content_view.label}-#{root_repo.label}"
+
+        if Katello::RootRepository.where(:label => root_repo.label).group(:label).count(:label)[root_repo.label] > 1
+          repo_query = Katello::Repository.joins(:root, :content_view_version => :content_view).
+              where("#{::Katello::RootRepository.table_name}.label" => root_repo.label, "#{::Katello::ContentView.table_name}.label" => content_view.label)
+          duplicate_count = repo_query.group(:library_instance_id, :content_view_id).count.select { |k,v| v > 1 }.count
+          name += "-#{root_repo.id}" if duplicate_count > 0
```

Comment:
I guess, this would be a breaking change to previous prepare runs, i.e. I suppose it would be mandatory to reset the migration and start a fresh prepare before you use this code. As far as I can tell, this name makes it into the pulpcore.core_repository table, column name. Thus, if you run another prepare with this change without reset it would add the already migrated repository again with the new name as well.

Reply:
yes, and for a few repos thats not a big deal.  I was trying to avoid renaming every single repo as this would nearly be a re-migration from scratch.